utils/delivery.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Delivery, DeliveryFrontend } from "@/types/delivery";
+
+export function parseDeliveryStructure(delivery: Delivery) {
+  let deliveryMap: DeliveryFrontend = {
+    id: delivery.id,
+    amount: delivery.amount,
+    deadline: delivery.deadline,
+    order: {
+      catagories: {},
+    },
+  };
+  delivery.orders.forEach((order) => {
```

Comment:
I've written a utility that converts deliveries to the desired format we discussed, maybe give it a look and see what you can use. It converts a list of Deliveries (Delivery[] -> Orders -> Amounts -> Items -> Categories) to:
```
{
  [category]: {
    [item]: Amount[]
  }
}
```

This is the code.
```
const deliveries = data.map((delivery) => {
    const amounts = delivery.orders.flatMap((o) => o.amounts);
    const categories = {} as Record<string, any>;

    for (const amount of amounts) {
      const { name: categoryName } = amount.item.category;
      const { name: itemName } = amount.item;

      const category = categories[categoryName] || {};
      const items = category[itemName] || [];
      items.push(amount);

      category[itemName] = items;
      categories[categoryName] = category;
    }

    return { ...delivery, categories };
  });
```

Reply:
Fixed